### ISSUE TITLE
After the upgrade process, there is no increase in the limits.

### ISSUE THREAD

**Issue by myavuzokumus** [Guillaume-Lebegue/IslandUpgrades#43]
### Expected behavior

Limits values ​​do not increase when we upgrade with the Upgrades plugin.

I would like to record a video but I don't have time.
Sorry if I'm posting this in the wrong section.

### Observed/Actual behavior

In the ./is limits menu, when we upgrade, the limit numbers should increase.

### Steps/models to reproduce

Upgrade in the Upgrade addon.
Then join after log out to the server and check the "/is limits" menu.

### BentoBox version

![image](https://user-images.githubusercontent.com/47111280/131114366-d0a07f51-1572-42a3-8a5b-bf54c1d9d7ca.png)

**Comment by tastybento** [comment]
@OverBrave The problem is that the author of Upgrades isn't active anymore, so we don't have an immediate person to fix this.

**Comment by tastybento** [comment]
I'll try and have a look and see.

**Comment by Guillaume-Lebegue** [comment]
Duplicate of #29

The thing with this problem is that upgrade overwrite limits permissions. However, if I fix it, it introduces a bug where the limits are reset each time the user logs back in.

So to fix it, I need a way to calculate the actual upgrade and with the actual implementation, It don't remember what was changed and I can't recalculate it. So I'm currently working on a total rework of the addon where I can easily calculate the actual upgrade as well as simplifying the configuration and pretty much everything #37.

I had a big stop in my development, but I restarted 2 days ago

**Comment by myavuzokumus** [comment]
> Duplicate of #29
> 
> The thing with this problem is that upgrade overwrite limits permissions. However, if I fix it, it introduces a bug where the limits are reset each time the user logs back in.
> 
> So to fix it, I need a way to calculate the actual upgrade and with the actual implementation, It don't remember what was changed and I can't recalculate it. So I'm currently working on a total rework of the addon where I can easily calculate the actual upgrade as well as simplifying the configuration and pretty much everything #37.
> 
> I had a big stop in my development, but I restarted 2 days ago

Not same issue. Please read again the issue. I didn't use any permission. I only upgraded any block in upgrade menu but nothing changed number in limit GUI and limit.

**Comment by tastybento** [comment]
@OverBrave noted - I can confirm - the limit doesn't increase.
@Guillaume-Lebegue I'm glad you're back!

Let's see how this goes. I'll still see if there's a quick fix, but it sounds like the idea is to do a rewrite.

**Comment by Guillaume-Lebegue** [comment]
> > Duplicate of #29
> > The thing with this problem is that upgrade overwrite limits permissions. However, if I fix it, it introduces a bug where the limits are reset each time the user logs back in.
> > So to fix it, I need a way to calculate the actual upgrade and with the actual implementation, It don't remember what was changed and I can't recalculate it. So I'm currently working on a total rework of the addon where I can easily calculate the actual upgrade as well as simplifying the configuration and pretty much everything #37.
> > I had a big stop in my development, but I restarted 2 days ago
> 
> Not same issue. Please read again the issue. I didn't use any permission. I only upgraded any block in upgrade menu but nothing changed number in limit GUI and limit.

Yes, however, someone asked me to allow limits permissions in and since then this bug is in.
I was pretty sure that I had put a fix to stop this bug and remove limits permissions. I'll look if I can do that real quick.

However, I would recommend against using this addon right now as I'm doing the rewrite and I don't think it will be compatible with the old version, or at least not directly.

**Comment by Guillaume-Lebegue** [comment]
I actually can't seem to find out why that doesn't work.

@tastybento I have a listener on LimitsJoinPermCheckEvent that I immediately cancel and the limits addon seems to properly fire it up (at least in code).

However, I can't debug it right now as I'm working with a totally different setup.

**Comment by ninjamaster4427** [comment]
I recently updated to Limits-1.17.2 and Upgrades-0.2.0-SNAPSHOT-b80. I have an issue where existing player data shows the player upgrades JSON are the same as pre-update while the actual limit (in the IslandBlockCount folder JSON data) is stuck at the default. For example, the default hopper limit is 10 and the example player is seeing /is limits for hoppers as 90 out of 10. They cannot place any more hoppers even though their limit should be 100 given their upgrade value which appears to be the same as pre-update.

Is this related to this Issue report or a separate issue?

**Comment by tastybento** [comment]
Maybe. I think it's fair to say it's broken right now.

**Comment by myavuzokumus** [comment]
What is the latest status, will it no longer be developed?
